benchmark/src/pre_benchmarks.cpp

```diff
@@ -0,0 +1,314 @@
+//==================================================================================
```

Comment:
This can't be written like the other benchmarks because this one needs the functions to be run in a sequence to be able to benchmark. The main(), usage() or security_model cannot be removed for this reason.

Reply:
Dmitry, we tried to alter this and wasted a day. It isn't a micro benchmark like any other. It is a full application example with benchmarking. If you don't want it, we will pull it and put it back in our application repo.